Orange/widgets/visualize/utils/widget.py

```diff
@@ -492,11 +492,12 @@ def setup_plot(self):
     def apply_selection(self):
         if self.data is not None and self.__pending_selection is not None \
                 and self.graph.n_valid:
-            index_group = [(index, group) for index, group in
-                           self.__pending_selection if index < len(self.data)]
+            index_group = [(i, group) for i, group in self.__pending_selection
+                           if i < self.graph.n_valid]
```

Comment:
Question: invalid indices appear when the current data is not the same as the one to which this selection applies. If this is the case, isn't it better to select nothing at all? Like

```
if self.data is not None and self.__pending_selection is not None \
    and max(i for i, _ in self.__pending_selection) < self.graph.n_valid:
```